### DISCUSSION TITLE
Using sonoff zigbee bridge

### DISCUSSION THREAD

**Discussion by jeroenst** [Koenkk/zigbee2mqtt#6450] [Devices]
I have a sonoff zigbee bridge flashed with tasmota zbbridge. I also want to put some of these on remote locations using mqtt over ssl.

Because more devices are supported in zigbee2mqtt and updating is easier on a server than on an embedded device, I would like to connect the sonoff zigbee bridge to zigbee2mqtt.

Is it possible to use the sonoff zigbee bridge as a dump gateway device which sends the raw zigbee data to zigbee2mqtt for further processing?

And if not is it possible to implement this?

![image](https://user-images.githubusercontent.com/2833940/109178567-47936900-7789-11eb-89cf-f9bcf4eceea9.png)
https://nl.aliexpress.com/item/32924140690.html

**Comment by Koenkk** [comment]
This sonoff zigbee bridge is currently not a supported adapter. There was some progress in https://github.com/Koenkk/zigbee-herdsman/issues/168 but this seems to have gone stale.

**Reply by new-kirte** [reply]
I am interested in sonoff bridge too. Or any other zigbee mesh extender, as it is needed in large house where the zigbee2mqtt is set up on one floor, but devices are deployed on second floor or across many brick walls.

**Comment by Hedda** [comment]
FYI, Zigbee Router firmware for EFR32MG21 adapters has now been released by xsp1989 on GitHub.

https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/Zigbee3.0_Dongle/RouterForDongle

From readme sounds as tested with ITead Zigbee 3.0 USB Dongle and SM-011 based USB adapters.

https://github.com/xsp1989/zigbeeFirmware/blob/710a60451f6631cad46f5d5107b65198558709c3/firmware/Zigbee3.0_Dongle/RouterForDongle/README.md

The same “SM-011 V1.0” Zigbee radio modules by CoolKit Technologies is also used inside some Zigbee gateways/hubs like this popular ITead Sonoff ZBBridge and ZB-GW03 eWeLink Ethernet Zigbee Gateway sold by EACHEN and SmartWise, so could perhaps be that the same Zigbee Router firmware could maybe also be used on the SM-011 Zigbee module in all or some of those products as well?

It might be that ITead Sonoff ZBBridge requires signed firmware image to work, see related discussion:

https://github.com/arendst/Tasmota/issues/9316

See related discussions here:

https://github.com/xsp1989/zigbeeFirmware/issues/16

https://github.com/xsp1989/zigbeeFirmware/issues/2

**Comment by Hedda** [comment]
Note! EZSP support is still in a very experimental phase so you should really post your feedback here -> https://github.com/Koenkk/zigbee-herdsman/issues/319

I am not sure if developer(s) is currently actively working on improving or maintaining that experimental support(?).

Only clear tips is to try upgrade EmberZNet NCP UART EZSP firmware Zigbee firmware version 6.7.9 or later and test dev branch (a.k.a. Edge):

https://github.com/arendst/Tasmota/tree/development/tools/fw_SonoffZigbeeBridge_ezsp

https://github.com/Koenkk/zigbee2mqtt/tree/dev

There looks to be some ezsp related fixes in the zigbee-herdsman dependency that might have not yet made it into Z2M dev/edge yet:

https://github.com/Koenkk/zigbee-herdsman/

Regardless you still want to direct most if not all feedback and issues to subscribers of the WIP discussion in the zigbee-herdsman repo:

https://github.com/Koenkk/zigbee-herdsman/issues/319

Also, please take note that WiFi based Zigbee bridges like ITead Sonoff ZBBridge are generally not recommended with the EZSP protocol. See the warnings about it:

https://www.zigbee2mqtt.io/guide/adapters/#notes

WiFi-based Serial-to-IP bridges are not recommended for Silicon Labs EZSP adapters as the serial protocol does not have enough fault-tolerance to handle packet loss or latency delays that can normally occur over WiFi connections. If cannot use a locally connected USB or UART/GPIO adapter then the recommendation is to use remote adapter that connected via Ethernet (wired) to avoid issues with EZSP caused by WiFi connections.

https://www.zigbee2mqtt.io/advanced/remote-adapter/connect_to_a_remote_adapter.html

WiFi-based Serial-to-IP bridges are not recommended for Silicon Labs EZSP adapters as the serial protocol does not have enough fault-tolerance to handle packet loss or latency delays that can normally occur over WiFi connections.

More warning about Zigbee to WiFi bridges for EZSP is not recommended for production as it does not do well with packed loss:

https://github.com/zigpy/bellows#warning-about-zigbee-to-wifi-bridges

https://www.home-assistant.io/integrations/zha/#warning-about-wi-fi-based-zigbee-to-serial-bridgesgateways

FYI, I also understand that the EFR32 board antenna design of the "SM-011 V1.0" Zigbee module used inside ITead's Sonoff ZBBridge have very high receive sensitivity to electromagnetic interference (EMI) / radio-frequency interference. This means that it will be extra important to keep it away from any other electronic devices or appliances that may generate an electromagnetic field (EM Field) or send our interfering radio signals. That includes the computer that you run your home automation application software on as well as any devices such as for example external harddrives or storage devices, power-supplies, etc., and of course any Wi-Fi access points or routers nearby.

Note! The new ITead Zigbee 3.0 USB Dongle Model: 9888010100045 (Hardware Revision Version 1.3) circuit board share the same type of antenna design so its radio is as well extremely sensitive to signal interference on its receiving end. The current recommendation for using it is to connect it via a long USB extension cable and try to move it as far away from all electronics and other possible sources of signal interference.

PS: Another general tip that always applies when building a Zigbee network is to begin by adding a few good mains-powered Zigbee router devices so that they will make use a stable baseline. Many start by buying three "IKEA TRADFRI Signal Repeater" (E1746).

https://zigbee.blakadder.com/Ikea_E1746.html

https://www.ikea.com/us/en/p/tradfri-signal-repeater-30400407/

https://www.ikea.com/gb/en/p/tradfri-signal-repeater-80424255/

https://www.ikea.com/se/sv/p/tradfri-signalfoerstaerkare-10400408/
